CORE-332: disable fence

davidangb opened the pull request:
Jira: https://broadworkbench.atlassian.net/browse/CORE-332

## What:
* Fence service account support is removed from ECM to stop calling Gen3
* ECM returns a 404 to any downstream services requesting a fence service account key
* Deleted (lots of) implementation code which previously supported fence keys

## Why:
Gen3 wants to stop supporting this feature on their end; therefore, ECM needs to stop calling Gen3.

## Notes:
I may have missed some implementation code which can still be deleted. Do you see anything else that should be deleted? This PR is already removing quite a lot.

davidangb committed:
fence API always returns 404

davidangb committed:
update openapi

davidangb committed:
fix test, cleanup

davidangb committed:
Merge branch 'dev' into da_CORE-332_disableFence

davidangb committed:
minor cleanup

davidangb committed:
remove FenceProviderService methods

davidangb committed:
delete FenceKeyRetriever

davidangb committed:
no need to revoke keys which we never return

davidangb committed:
remove more unused classes

davidangb committed:
remove FenceAccountKeyDAO

davidangb committed:
remove unused variables

davidangb committed:
remove TestUtils.createRandomFenceAccountKey

davidangb committed:
delete model

davidangb committed:
clean up test

dvoet commented on service/src/main/java/bio/terra/externalcreds/services/FenceProviderService.java:
```diff
@@ -19,32 +18,22 @@
 @Slf4j
 public class FenceProviderService extends ProviderService {
```
I think this class can go away and use `TokenProviderService` instead.

Bonus points: merge `TokenProviderService` with `PassportProviderService`, remove the big switch statement in `OauthApiController.createLink`, drive the decision about whether or not to get a passport from either a new config setting for a provider or from the existence of `ga4gh_passport_v1` scope.

dvoet reviewed (changes requested):
1 inline comment

also in service/src/main/resources/providers.yml, remove instances of `google_credentials` scope

davidangb committed:
delete FenceProviderService entirely

davidangb committed:
simplify FenceAccountKeyApiControllerTest

davidangb committed:
remove google_credentials scope

davidangb commented:
@dvoet I have removed `google_credentials` scopes in provider config, and removed `FenceProviderService` entirely in favor of `TokenProviderService`.

I am game to tackle merging `TokenProviderService` with `PassportProviderService` … but how about doing that in a separate PR, since this one is already pretty big?
